src/cpu/aarch64/acl_inner_product.hpp

```diff
@@ -98,6 +101,7 @@ struct acl_inner_product_fwd_t : public primitive_t {
                             primitive_attr_t::skip_mask_t::post_ops, f32);
             const bool ok = is_fwd() && !has_zero_dim_memory()
                     && utils::one_of(true, is_fp16_ok, is_fp32_ok)
+                    && weights_md_.format_kind == format_kind::any
```

Comment:
Yes.

I thought that using the `format::any` is best practice, and we expected that most users do this already, particularly for inference. This is certainly the case for TensorFlow, where this will allow the reorders to be hoisted out. Do you know whether users tend to do this?

Am I right in thinking that to support specific formats, we would need to implement the reorder logic inside each primitive?

Reply:
> Yes.
> 
> I thought that using the `format::any` is best practice, and we expected that most users do this already, particularly for inference. This is certainly the case for TensorFlow, where this will allow the reorders to be hoisted out. Do you know whether users tend to do this?


You are right, the best practice and a programming model tells user to create inner product with `any` so oneDNN chooses the best implementation. However sometimes it is not possible on user's side (when weights are constant and can't be reordered by a framework) and oneDNN programming model supports passing plain layouts to inner product -- it will not return unimplemented/invalid arguments. After your changes inner product created with plain weights layout will dispatch into gemm-based or reference inner product which is totally fine, but might result in worse performance comparing to previos state. 

> Am I right in thinking that to support specific formats, we would need to implement the reorder logic inside each primitive?

This is a possible solution, but there are implications:
- scratchpad is required because primitive is stateless so we can't cache changes inside
- reorder will be required each execution due to stateless primitive


Again, no changes are required on your side, this is completely fine from oneDNN programming model I just wanted to make sure this is your intention (disabling explicit weight formats in acl ip implementation). You can resolve this discussion.